src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -929,6 +970,44 @@ else if (FreeMode.SUPPLYCAMP == Settings.instance.getFreeMode())
         }
     }
 
+    private String blockListToCommaSeparatedString(List<BlockPos> blocksToReportList) {
+        StringBuilder outputListStringBuilder = new StringBuilder();
+        boolean firstItem = true;
+        for (BlockPos blockPos : blocksToReportList) {
+            if (firstItem) {
+                firstItem = false;
+            } else {
+                outputListStringBuilder.append(',');
+                outputListStringBuilder.append(' ');
+            }
+            outputListStringBuilder.append('(');
+            outputListStringBuilder.append(blockPos.getX());
+            outputListStringBuilder.append(' ');
+            outputListStringBuilder.append(blockPos.getY());
+            outputListStringBuilder.append(' ');
+            outputListStringBuilder.append(blockPos.getZ());
+            outputListStringBuilder.append(')');
+        }
+        String outputList = outputListStringBuilder.toString();
+        return outputList;
+    }
+
+    private Map<PlacementErrorType, List<BlockPos>> partitionPlacementErrorsByErrorType(
+            List<PlacementError> placementErrorList) {
+        Map<PlacementErrorType, List<BlockPos>> blockPosListByErrorTypeMap = new HashMap<PlacementErrorType, List<BlockPos>>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert this Map to an EnumMap. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1640)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)